src/App.php

```diff
@@ -15,17 +15,30 @@ public function isDownForMaintenance(): bool
     }
 
     /**
-     * @param  string|array  $environments
+     * @param string|string[] $environments
      * @return string|bool
      *
      * @see \Illuminate\Contracts\Foundation\Application::environment()
      */
     public function environment(...$environments)
     {
-        if(!empty($environments)) {
-            return in_array('torch', $environments);
+        if(empty($environments)) {
+            return 'torch';
         }
 
-        return 'torch';
+        return in_array(
+            'torch',
+            is_array($environments[0]) ? $environments[0] : $environments
```

Comment:
Just for my curiosity, in what case would we be passing an array whose first index would be an array of environments? Thanks!

Reply:
It's because of the `...` opperator.
```php
environment('a', 'b') => ['a', 'b']
environment(['a', 'b']) => [['a', 'b']]
```
This part is copied from the illuminate core - I think if we have this example class it should be fully compatible with the interface - even if we don't implement it as we don't need all methods.

Reply:
👍  Sounds good, thanks!